src/main/java/org/jabref/gui/fieldeditors/UrlEditor.java

```diff
@@ -23,6 +28,15 @@ public UrlEditor(String fieldName, DialogService dialogService, AutoCompleteSugg
         ControlHelper.loadFXMLForControl(this);
 
         textArea.textProperty().bindBidirectional(viewModel.textProperty());
+        Supplier<List<MenuItem>> contextMenuSupplier = EditorMenus.getCleanupURLMenu(textArea);
+        textArea.addToContextMenu(contextMenuSupplier);
+        MenuItem cleanupURLMenuItem = contextMenuSupplier.get().get(0);
```

Comment:
Thats a bit ugly: you need to know that the cleanup url menu item is the first one returned. Thus the code fails if somebody changes the `getCleanupURLMenu` method. Everything related to the context-menu should actually go there. So much for the theory; I'm not sure if and how a menu item can be notified that it is displayed. If you don't find a good solution, I would remove the disable check for the moment.